### ISSUE TITLE
gnucash: No suitable backend for sqlite save files

### ISSUE THREAD

**Issue by nathantalewis** [NixOS/nixpkgs#57445]
## Issue description
Using gnucash-3.2, I am trying to open a save file that is in the sqlite format. I get a `No suitable backend was found for <filename>.` error. Also, when going to "Save as..." from the File menu, the only Data Format available in the dropdown is `xml`.

### Steps to reproduce
Open gnucash, go to File > Save As..., try to select the sqlite data format (or any data format other than xml).

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.95, NixOS, 18.09.1985.749a3a0d00b (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(<username>): `"nixos-unstable-19.09pre172038.5d3fd3674a6"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by au-phiware** [comment]
I am trying to connect to a mysql database with gnucash-3.8b and I get an `No suitable backend was found for mysql://...` error.  Running with `--debug` I see:

```
* 15:05:49  INFO <qof.session  > [QofSessionImpl::load_backend] The provider providers access_method, file, but we're loading for access_method, mysql. Skipping.
```

I tried adding `libdbiDrivers` to the `buildInputs` but to no avail:

```
      (gnucash.overrideAttrs (oldAttrs: {
        buildInputs = oldAttrs.buildInputs ++ [ libdbiDrivers ];
      }))
```

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.39, NixOS, 20.03.1822.5adf2a6c116 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-20.03.1822.5adf2a6c116"`
 - channels(corin): `"nixpkgs-20.09pre225085.cfe68f2b68b"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Please advise.

**Comment by nathantalewis** [comment]
I never found a way to fix the issue, so I used GnuCash on another machine to convert my files to 'xml' from 'mysql'.

**Comment by au-phiware** [comment]
I've ended up installing gnucash via flatpak.  I would love to know how to do it with nix but I just can't get my head around cmake at the moment 😕

**Comment by jasom** [comment]
Here's a patch that fixes the problem; I ran into it independently yesterday.  If nobody else submits a PR, I'll get setup for that.
```
--- default.nix 2020-05-20 09:13:28.963013780 -0700
+++ /nix/store/lp3xw6fildch606zvqm16nvq0jq14q7d-nixos-20.03.1589.9137f05564e/nixos/pkgs/applications/office/gnucash/default.nix 1969-12-31 16:00:01.000000000 -0800
@@ -65,7 +65,6 @@
       --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH:$out/share/gsettings-schemas/${pname}-${version}" \
       --prefix XDG_DATA_DIRS : "${hicolor-icon-theme}/share" \
       --prefix PERL5LIB ":" "$PERL5LIB" \
-      --set GNC_DBD_DIR ${libdbiDrivers}/lib/dbd \
       --prefix GIO_EXTRA_MODULES : "${stdenv.lib.getLib dconf}/lib/gio/modules"
   '';
 ```

**Comment by jasom** [comment]
This appears to be fixed by #88442
